File: Practice Sorting/15_Counting sort.py
def frequency_based_sort(n, arr):
    # Write your code here
    min_value = min(arr)
    max_value = max(arr)

    frequency = [0] * (max_value - min_value + 1)

    for num in arr:
        frequency[num - min_value] += 1

    sorted_arr = []
    for i in range(n):
        val=arr[i]
        sorted_arr.append((frequency[val - min_value], val))

    sorted_arr.sort()

    for pair in sorted_arr:
        print(pair[1], end=" ")

if __name__ == "__main__":
    n = int(input())
    arr = list(map(int, input().split()))

    # Function call to perform frequency-based sorting
    frequency_based_sort(n, arr)

File: Practice Sorting/codigo_cs/15_Counting sort.cs
using System;
using System.Linq;
using System.Collections.Generic;

public class Test15
{



    static string Aux(int n, int[] arr)
    {
        int min_value = arr.Min();
        int max_value = arr.Max();

        int[] frequency = new int[max_value - min_value + 1];

        for (int i = 0; i < n; i++)
        {
            frequency[arr[i] - min_value]++;
        }
        var sortedArr = new List<(int freq, int val)>();
        for (int i = 0; i < n; i++)
        {
            sortedArr.Add((frequency[arr[i] - min_value], arr[i]));
        }
        sortedArr.Sort();
        return string.Join(" ", sortedArr.Select(t => t.val));

    }
    public static void CountingSort()
    {
        int n = int.Parse(Console.ReadLine());
        int[] arr = Array.ConvertAll(Console.ReadLine().Split(' '), int.Parse);

        Console.WriteLine(Aux(n, arr));

    }
}
